sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSender.cs

```diff
@@ -405,9 +405,9 @@ private DiagnosticScope CreateDiagnosticScope(IEnumerable<ServiceBusMessage> mes
             _connection.ThrowIfClosed();
             cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
 
-            IReadOnlyList<ServiceBusMessage> messageList = messages switch
+            List<ServiceBusMessage> messageList = messages switch
```

Comment:
I guess one drawback the change from `IReadOnlyCollection` to `List` has is that previously if someone passed an array or a list we could directly pass it through, since both implement `IReadOnlyCollection` if I'm not mistaken. Now for all calls where the enumerable is an array we would allocate a list. Judging by that I think the original IReadOnlyCollection approach I had was smarter. Can you double-check my assumptions?

Reply:
I double-checked and indeed List and Array both provide IReadOnlyCollection. Therefore using that as the inner type seems smarter because then we don't start allocating for probably the cases when users pass an array as an enumerable. So I switched back in commit https://github.com/Azure/azure-sdk-for-net/pull/26911/commits/19cf8fc0c47c520b453b02b75be9be8cbfeaf689